File: HelpfulSense/app/src/main/java/com/andruid/magic/helpfulsense/viewmodel/ActionViewModel.java
package com.andruid.magic.helpfulsense.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.andruid.magic.helpfulsense.model.Action;
import com.andruid.magic.helpfulsense.repo.ActionRepository;
import com.andruid.magic.helpfulsense.service.UpdateIntentService;

import java.util.List;

import timber.log.Timber;

public class ActionViewModel extends AndroidViewModel {
    private MutableLiveData<List<Action>> actionLiveData;

    public ActionViewModel(@NonNull Application application) {
        super(application);
        Timber.d("created view model");
        actionLiveData = ActionRepository.getInstance()
                .getActionsFromFile(application.getApplicationContext());
    }

    public LiveData<List<Action>> getSavedActions() {
        Timber.d("get live data from view model");
        return actionLiveData;
    }

    public void updateSavedActions(List<Action> actions){
        actionLiveData.setValue(actions);
        UpdateIntentService.startActionUpdate(getApplication(), actions);
    }
}

File: HelpfulSense/app/src/main/java/com/andruid/magic/helpfulsense/ui/viewmodel/ActionViewModel.kt
package com.andruid.magic.helpfulsense.ui.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.asLiveData
import androidx.lifecycle.liveData
import com.andruid.magic.helpfulsense.database.DbRepository
import com.andruid.magic.helpfulsense.database.entity.Action

/**
 * ViewModel for getting all [Action] as [LiveData]
 */
class ActionViewModel : ViewModel() {
    val actionLiveData = liveData {
        emitSource(DbRepository.fetchAllActions().asLiveData())
    }
}